text/0059-token-metadata-document.md

```diff
@@ -0,0 +1,485 @@
+
+Clients MAY continue to bake token metadata into their build at their discretion, for instance, in order to support offline wallets or other cases where the client cannot access the internet. When both baked-in metadata and the `token_metadata.json` are available, the validated `token_metadata.json` should be considered a more reliable source of truth.
+
+Maintainers of this document SHOULD NOT:
+
+* Allow duplicate records for a given token id
+* Delete a token id's record
+* Modify data such as the `ticker_symbol` or `decimals` of a token, which may confuse users, and particularly, any exchanges that use this as a source of truth.
+* There may be legitimate reasons to make changes the `currency_name` or `logo`, but this needs to be carefully considered.
+* Allow two token ids with `currency_name`s which could be confused
+* Allow the same `ticker_symbol` to appear, or, allowing two `ticker_symbols` which differ only in case, or the replacement of letters with similar numbers.
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+This proposal has a few drawbacks relative to [MCIP 40](https://github.com/mobilecoinfoundation/mcips/pull/0040):
+
+* MCIP 40 puts the token metadata on chain. This makes it possible to enforce protocol level rules around how the document can evolve. It also ensures the availability of the document.
+* It also would take longer to develop since it would require an enclave upgrade.
+
+This proposal's authentication scheme is a single ed25519 signature.
+
+* As an alternative, we could have a signature chain, such as an x509. This allows for more sophisticated strategies around the keeping the key secure.
+* As an alternative, we could have a k-of-n ed25519 multisignature. This would allow for the signing authority to be shared across several parties who can sign asynchronously.
+
+Since this signature can easily be performed off-line anyways, and the document isn't expected to change often, we think a single ed25519 signature is fine for this stop-gap proposal.
+
+In our design, we included SVG images inline in the metadata document to avoid this validation issue.
+This will not scale well to e.g. millions of tokens, and we expect to transition to on-chain metadata before that becomes a problem.
+
+This proposal does not envision having different token registries per network. It is believed that this complexity is unnecessary.
+However, it means that just as clients need to hit Intel's IAS services as part of testing (in prod and dev), they will have to hit
+the `mobilecoin.com` URLs as well, so this is another way in which testing a deployed network will not be completely isolated to the servers deployed as part of the network.
```

Comment:
We could also have the consensus nodes serve this file and signature if we want to.

Reply:
Or equivalently, have a flag controlling the host used to fetch the token metadata.

Reply:
came here to discuss having the consensus nodes serve the file as well. Do our nodes already have facilities to serve out files? It would remove a point of central failure for clients to distribute the token metadata to all of the nodes.

Reply:
The consensus nodes don't have to serve the file over HTTP. Since they speak gRPC, we can define an RPC that returns the metadata JSON and signature.